app/src/main/java/com/igordotsenko/dotsenkorssreader/ReaderContentProvider.java

```diff
@@ -0,0 +1,168 @@
+                throw new IllegalArgumentException("Wrong URI: " + uri);
+        }
+
+        establishDatabaseConnection();
+        rowID = mDatabaseConnection.insert(tableName, null, values);
+        resultUri = ContentUris.withAppendedId(contentUri, rowID);
+        getContext().getContentResolver().notifyChange(uri, null);
+        return resultUri;
+    }
+
+    @Override
+    public boolean onCreate() {
+        try {
+            mDbHandler = new DBHandler(
+                    getContext(), MainActivity.DB_NAME, null, MainActivity.DB_VERSION);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public Cursor query(
+            Uri uri,
+            String[] projection,
+            String selection,
+            String[] selectionArgs,
+            String sortOrder) {
+
+        String tableName;
+
+        switch ( uriMatcher.match(uri) ) {
+            case CHANNEL:
+                tableName = ContractClass.CHANNEL_TABLE;
+                break;
+            case ITEM:
+                tableName = ContractClass.ITEM_TABLE;
+                break;
+            default:
+                throw new IllegalArgumentException("Wrong URI: " + uri);
+        }
+
+        establishDatabaseConnection();
+        Cursor cursor = mDatabaseConnection.query(
+                tableName, projection, selection, selectionArgs, null, null, sortOrder);
+
+        cursor.setNotificationUri(
+                getContext().getContentResolver(), ContractClass.ITEM_CONTENT_URI);
+
+        return cursor;
+    }
+
+    @Override
+    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
+        String tableName;
+        int rowsUpdatedCount;
+
+        switch ( uriMatcher.match(uri) ) {
+            case CHANNEL:
+                tableName = ContractClass.CHANNEL_TABLE;
+                break;
+            case ITEM:
+                tableName = ContractClass.ITEM_TABLE;
+                break;
+            default:
+                throw new IllegalArgumentException("Wrong URI: " + uri);
+        }
+
+        establishDatabaseConnection();
+        rowsUpdatedCount = mDatabaseConnection.update(tableName, values, selection, selectionArgs);
+        getContext().getContentResolver().notifyChange(uri, null);
+
+        return rowsUpdatedCount;
+    }
+
+    private void establishDatabaseConnection() {
+        if ( mDatabaseConnection == null ) {
+            try {
+                mDatabaseConnection = mDbHandler.getDatabaseConnection();
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new Error("Error during database connection establishing");
+            }
+        }
+    }
+
+    public static class ContractClass {
+        public static final String AUTHORITY = "com.igordotsenko.dotsenkorssreader";
+
+        public static String CHANNEL_TABLE = Channel.TABLE;
```

Comment:
Better separate constants in Contract class by entries.
It's better for readability and understanding.

For example: 

class Contract {
   public static class Channel {
   // all constants, uris and helper methods  for channel 
   }

    public static class Item {
      // all constants, uris and helper methods  for item 
      }
}